### PR TITLE
Rework of date plugin with custom date NodeSpec

### PR DESCRIPTION
**Pull request by elpoelma**

### CODE REVIEW

**Review by lagartoverde — CHANGES_REQUESTED** [review]
Works good, I think we are missing 2 things, first of all the date gets setted to the actual time it's now, we should set the hour, minutes, seconds and ms to 0 if we are not specifing them in the modal. Second we have to modify the variable plugin to insert this new node :)

**Comment by elpoelma** [comment]
> Works good, I think we are missing 2 things, first of all the date gets setted to the actual time it's now, we should set the hour, minutes, seconds and ms to 0 if we are not specifing them in the modal. Second we have to modify the variable plugin to insert this new node :)

Ok I will be looking into reworking the insert-variable plugin configuration in order to support this.

File: addon/plugins/rdfa-date-plugin/nodes/date.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    },
+  },
+  selectable: true,
+  toDOM: (node) => {
+    const { value, onlyDate } = node.attrs;
+    const datatype = onlyDate ? XSD('date') : XSD('dateTime');
+    const humanReadableDate = value
+      ? formatDate(new Date(value), onlyDate)
+      : onlyDate
+      ? 'Voeg datum in'
```
Comment (abeforgit) [inline addon/plugins/rdfa-date-plugin/nodes/date.ts:26]: Would be good to i18n this

File: addon/components/rdfa-date-plugin/insert.ts [hunk 2]
```diff
@@ -16,29 +16,12 @@ export default class RdfaDatePluginInsertComponent extends Component<Args> {
...
     this.controller.withTransaction((tr) => {
       return tr.replaceSelectionWith(
-        this.schema.text('${date and time}', [
-          this.schema.mark('inline_rdfa', {
-            datatype: 'xsd:dateTime',
-            property: 'ext:content',
-          }),
-        ]),
-        false
+        this.schema.node('date', {
```
Comment (abeforgit) [inline addon/components/rdfa-date-plugin/insert.ts:22]: Would be nice to autoselect the inserted date here I think

**Review by abeforgit — CHANGES_REQUESTED** [review]
Note I think we have to add format support before we can release this to prod, will make a jira story

**Review by abeforgit — APPROVED** [review]

**Pull request merged**